scheduler/prioritize.go

```diff
@@ -17,6 +17,13 @@ func capacityToScore(capacity uint64, divisor float64) int {
 
 	// avoid logarithm of zero, which diverges to negative infinity.
 	if gb == 0 {
+		// if there is a non-nil capacity but we dont have at least one gigabyte, we score it with one.
+		// this is because the capacityToScore precision is at the gigabyte level.
+		// TODO: introduce another scheduling algorithm for byte-level precision.
+		if capacity > 0 {
+			return 1
+		}
```

Comment:
@jakobmoellerdev @toshipp 
When capacity is between 1 byte and 1GiB, I noticed there is a the following is a curious reversal.
```
capacity           score
0                  0
1Byte ~ 1GiB       1
1GiB ~ 2GiB        0
2GiB ~             1~
(default divisor : 1)
```

I have two ideas to solve this problem, and I think idea 1 is better. what do you think?

1.  Return 1 as a score when capacity != 0 && converted < 1.
2.  Return 0 as a score when capacity < 1GiB.

Reply:
> 1. Return 1 as a score when capacity != 0 && converted < 1.

do you mean converted = size >> 30 ?

Because I think this first case is what happens right? 

When we return 0 as score when capacity <1GiB then we have an issue with scoring nodes because nodes without capacity are on the same score right?

Reply:
I do think your issue is valid though, basically as long as capacity > 0, score must be at least 1.

Reply:
The only way I could think of to solve this would be to add special cases to all statements that could return 0 if capacity > 0, also in the after the calculating log2

Reply:
@llamerada-jp @toshipp I think this could solve the issue. I had to introduce one more edge case as sometimes the logarithm can also result in score 0, so i just bounded it there as well. This also makes it safe against any divisor value
```go
func capacityToScore(capacity uint64, divisor float64) int {
	gb := capacity >> 30

	// Avoid logarithm of zero, which diverges to negative infinity.
	if gb == 0 {
		// If there is a non-nil capacity but we dont have at least one gigabyte, we score it with one.
		// This is because the capacityToScore precision is at the gigabyte level.
		// TODO: introduce another scheduling algorithm for byte-level precision.
		if capacity > 0 {
			return 1
		}

		return 0
	}

	converted := int(math.Log2(float64(gb) / divisor))
	score := converted
	switch {
	case score < 0:
		score = 0
	case score > 10:
		score = 10
	}
	
	// Also return score 1 here if we retrieve a score of 0 but we have a non-0 capacity.
	if capacity > 0 && score == 0 {
		return 1
	}
	
	return score
}
```

Reply:
When capacity == 0, gb becomes 0 and is returned before the switch statement.
So will the switch statement be like below?
```go
	converted := int(math.Log2(float64(gb) / divisor))
	switch {
	case converted < 1:
		return 1
	case converted > 10:
		return 10
	default:
		return converted
	}
```

Reply:
Adjusted accordingly and all tests still pass.